ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/caching/ImmutablesOnlyCacheUpdatePolicySpec.scala

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator.caching
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlStateKey
+import com.daml.ledger.validator.TestHelper
+import com.daml.ledger.validator.caching.ImmutablesOnlyCacheUpdatePolicy._
+import org.scalatest.{Matchers, WordSpec}
+
+class ImmutablesOnlyCacheUpdatePolicySpec extends WordSpec with Matchers {
+  "cache update policy" should {
+    "allow write-through caching of packages" in {
+      shouldCacheOnWrite(aPackageKey) shouldBe true
```

Comment:
If we're testing against the interface, I'd prefer if we set `val policy = ImmutablesOnlyCacheUpdatePolicy` and then checked:

```suggestion
      policy.shouldCacheOnWrite(aPackageKey) shouldBe true
```

Reply:
Good point -- changed accordingly.